File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    f, t, h = 0, 0, 0
    first = [1, 2, 3, 4, 5]
    two = [2, 1, 2, 3, 2, 4, 2, 5]
    three = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    for i in range(len(answers)):
        c1 = i % 5
        c2 = i % 8
        c3 = i % 10

        if first[c1] == answers[i]:
            f += 1
        if two[c2] == answers[i]:
            t += 1
        if three[c3] == answers[i]:
            h += 1
    
    m = max(f, t, h)
    res = []

    if m == f:
        res.append(1)
    if m == t:
        res.append(2)
    if m == h:
        res.append(3)
        
    return res

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] first = new int[]{1, 2, 3, 4, 5}; // 5
        int[] second = new int[]{2, 1, 2, 3, 2, 4, 2, 5};
        int[] third = new int[]{3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        Map<Integer, Integer> score = new HashMap<>();
        score.put(1, 0);
        score.put(2, 0);
        score.put(3, 0);
        
        for(int i=0; i<answers.length; i++) {
            int answer = answers[i];
            int aF = first[i % first.length];
            int aS = second[i % second.length];
            int aT = third[i % third.length];
            
            if(aF == answer) {
                score.put(1, score.get(1) + 1);
            }
            if(aS == answer) {
                score.put(2, score.get(2) + 1);
            }
            if(aT == answer) {
                score.put(3, score.get(3) + 1);
            }
        }
        
        int maxScore = Math.max(score.get(1), Math.max(score.get(2), score.get(3)));
        
        List<Integer> numbers = new ArrayList<>();
        for(int i=1; i<=3; i++) {
            if(score.get(i) == maxScore) {
                numbers.add(i);
            }
        }
        return numbers.stream().mapToInt(i -> i).toArray();
    }
}